src/lib/Libwin/passwd.c

```diff
@@ -4814,20 +4814,21 @@ logon_pw(char *username,
 	size_t credl,
 	int (*decrypt_func)(char *, int, size_t, char **),
 	int use_winsta,
-	char msg[4096])
+	char *msg)
 {
-	static  char	*id = "logon_pw";
-	int		i;
-	struct	passwd *pwdp;
-	SID	*usid;
-	char	*homedir;
-	char  	thepass[PWLEN+1];
-	char 	realname[UNLEN+1];
-	char	domain[PBS_MAXHOSTNAME+1] = "";
-	char	*p;
-	char	msg2[4096];
+	int		i = 0;
+	struct	passwd *pwdp; = NULL
+	SID	*usid = NULL;
+	char	*homedir = NULL;
+	char  	thepass[PWLEN+1] = {'\0'};
+	char 	realname[UNLEN+1] = {'\0'};
+	char	domain[PBS_MAXHOSTNAME+1] = {'\0'};
+	char	*p = {'\0'};
+	char	msg2[LOG_BUF_SIZE] = {'\0'};
```

Comment:
Now you need to find each place where logon_pw() is called and ensure the last parameter being passed is also defined using LOG_BUF_SIZE. For example, in pbs_runas_win.c the macro CMDLINE_LEN is used to define msg. In some cases you may need to include the log.h header. Make sure you do a test build to make sure it all compiles.